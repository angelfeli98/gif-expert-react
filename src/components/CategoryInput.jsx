
import { useState } from 'react';
import PropTypes from 'prop-types';

function CategoryInput({onNewValue}) {

    const [value, setValue] = useState('');

    const handlKeyUp = (e) => {
        if(e.key === 'Enter' && value.trim() !== '') {
            onNewValue(value);
            setValue('');
        }
    }

    return (
        <input
            type="text"
            placeholder="Categoria"
            onKeyUp={handlKeyUp}
            value={value}
            onChange={(e) => setValue(e.target.value)}
        />
    )
}

CategoryInput.propTypes = {
    onNewValue: PropTypes.func.isRequired
};

export default CategoryInput;