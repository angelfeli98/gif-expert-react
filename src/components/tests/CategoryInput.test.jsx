/* eslint-disable no-undef */
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import CategoryInput from '../CategoryInput';

describe('<CategoryInput />', () => {

    let onNewValue;

    beforeEach(() => onNewValue = jest.fn());

    test('should render correctly', () => {
        const { container } = render(<CategoryInput onNewValue={onNewValue} />);
        expect(container).toMatchSnapshot();
    })

    test('should change value', () => {
        render(<CategoryInput onNewValue={onNewValue} />);
        const input = screen.getByRole('textbox');
        fireEvent.change(input, {target: {value: 'Morty'}});
        expect(input).toHaveValue('Morty');
    })

    test('should set new value', () => {
        render(<CategoryInput onNewValue={onNewValue} />);
        const input = screen.getByRole('textbox');
        fireEvent.change(input, {target: {value: 'Morty'}});
        fireEvent.keyUp(input, {key: 'Enter'});
        expect(onNewValue).toHaveBeenCalledWith('Morty');
    })

    test('should not set new value', () => {
        render(<CategoryInput onNewValue={onNewValue} />);
        const input = screen.getByRole('textbox');
        fireEvent.change(input, {target: {value: ''}});
        fireEvent.keyUp(input, {key: 'Enter'});
        expect(onNewValue).not.toHaveBeenCalled();
    })
})