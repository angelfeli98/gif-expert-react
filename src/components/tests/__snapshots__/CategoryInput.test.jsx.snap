// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryInput /> should render correctly 1`] = `
<div>
  <input
    placeholder="Categoria"
    type="text"
    value=""
  />
</div>
`;
